packages/sit-onyx/src/utils/playwright.tsx

```diff
@@ -103,35 +111,54 @@ export const createScreenshotsForAllStates =
     states: S,
     baseName: string,
     caseBuilder: CaseBuilder<S>,
+    options?: CreateScreenshotsForAllStatesOptions,
   ) =>
-  async ({ mount, page }: TestArg) => {
+  async ({ mount, page, makeAxeBuilder }: TestArg) => {
+    /* eslint-disable playwright/no-standalone-expect */
+
+    const axeBuilder = makeAxeBuilder();
+    const performAxeScan = options?.disableAxeCheck
+      ? () => Promise.resolve()
+      : async () => axeBuilder.analyze().then((r) => expect(r.violations).toEqual([]));
+
     const permutations = generatePermutations(states);
 
+    // give a maximum of 1 second per test case
+    test.setTimeout(permutations.length * 1000);
+
     for (const testCase of permutations) {
-      // ARRANGE
-      const wrappedMount = ((jsx, options) =>
-        mount(
-          <div
-            style={{ width: "max-content", padding: "1rem" }}
-            class={{
-              "onyx-use-optional": options?.useOptional,
-            }}
-          >
-            {jsx}
-          </div>,
-        )) satisfies WrappedMount;
-
-      await page.getByRole("document").focus(); // reset focus
-      await page.getByRole("document").hover(); // reset mouse
-      await page.mouse.up(); // reset mouse
-      const component = await caseBuilder(testCase, wrappedMount, page);
-
-      // ASSERT
       const screenshotName = [
         baseName,
         ...Object.entries(testCase).map(([key, value]) => `${key}--${value}`),
       ].join("-");
-      await expect(component).toHaveScreenshot(`${screenshotName}.png`);
+
+      await test.step(screenshotName, async () => {
+        // ARRANGE
+        const wrappedMount = ((jsx, options) =>
+          mount(
+            <div
+              style={{ width: "max-content", padding: "1rem" }}
+              class={{
+                "onyx-use-optional": options?.useOptional,
+              }}
+            >
+              {jsx}
+            </div>,
+          )) satisfies WrappedMount;
+
+        await page.getByRole("document").focus(); // reset focus
+        await page.getByRole("document").hover(); // reset mouse
+        await page.mouse.up(); // reset mouse
+
+        // Setup component test case
+        const component = await caseBuilder(testCase, wrappedMount, page);
+
+        // ASSERT
+        const screenshotResult = expect(component).toHaveScreenshot(`${screenshotName}.png`);
+        const accessibilityScanResult = performAxeScan();
+
+        await Promise.all([screenshotResult, accessibilityScanResult]);
```

Comment:
Is it safe to run them in parallel? I remember some firefox issues in the past where the axe tests would cause a page reload

Reply:
Locally I don't have any issues, lets see what the CI says... A page reload would be destroy any component state and browser state, sounds like something that shouldn't happen. But yeah could very well be that Playwright isn't intended to be used like this.